src/lib/LBT/LBT.cpp

```diff
@@ -0,0 +1,153 @@
+      case SX1280_LORA_SF6: return 141;
+      case SX1280_LORA_SF7: return 218;
+      case SX1280_LORA_SF9: return 218;
+      default: return 218; // Values above 100mW are not relevant, default to 100mW threshold
+    }
+}
+
+int8_t ICACHE_RAM_ATTR PowerEnumToLBTLimit(PowerLevels_e txPower)
+{
+  // Calculated from EN 300 328, adjusted for 800kHz BW for sx1280
+  // TL = -70 dBm/MHz + 10*log10(0.8MHz) + 10 × log10 (100 mW / Pout) (Pout in mW e.i.r.p.)
+  // This threshold should be offset with a #define config for each HW that corrects for antenna gain,
+  // different RF frontends.
+  // TODO: Maybe individual adjustment offset for differences in
+  // rssi reading between bandwidth setting is also necessary when other BW than 0.8MHz are used.
+
+  switch(txPower)
+  {
+    case PWR_10mW: return -61 + LBT_RSSI_THRESHOLD_OFFSET_DB;
+    case PWR_25mW: return -65 + LBT_RSSI_THRESHOLD_OFFSET_DB;
+    case PWR_50mW: return -68 + LBT_RSSI_THRESHOLD_OFFSET_DB;
+    case PWR_100mW: return -71 + LBT_RSSI_THRESHOLD_OFFSET_DB;
+    // Values above 100mW are not relevant, default to 100mW threshold
+    default: return -71 + LBT_RSSI_THRESHOLD_OFFSET_DB;
+  }
+}
+
+void ICACHE_RAM_ATTR BeginClearChannelAssessment(void)
+{
+  if (!LBTEnabled)
+  {
+    return;
+  }
+
+  if(!LBTStarted)
+  {
+    Radio.RXnb();
+    rxStartTime = micros();
+    LBTStarted = true;
+  }
+}
+
+bool ICACHE_RAM_ATTR ChannelIsClear(void)
+{
+  if (!LBTEnabled)
+  {
+    return true;
+  }
+  LBTSuccessCalc.inc(); // Increment count for every channel check
+
+  LBTStarted = false;
+
+  // Read rssi after waiting the minimum RSSI valid delay.
+  // If this function is called long enough after RX enable,
+  // this will always be ok on first try as is the case for TX.
+
+  // TODO: Better way than busypolling this for RX?
+  // this loop should only run for RX, where listen before talk RX is started right after FHSS hop
+  // so there is no dead-time to run RX before telemetry TX is supposed to happen.
+  // if flipping the logic, so telemetry TX happens right before FHSS hop, then TX-side ends up with polling here instead?
+  // Maybe it could be skipped if there was only TX of telemetry happening when FHSShop does not happen.
+  // Then RX for LBT could stay enabled from last received packet, and RSSI would be valid instantly.
+  // But for now, FHSShops and telemetry rates does not divide evenly, so telemetry will some times happen
+  // right after FHSS and we need wait here.
+
+  int validRSSIdelayUs = SpreadingFactorToRSSIvalidDelayUs((SX1280_RadioLoRaSpreadingFactors_t)ExpressLRS_currAirRate_Modparams->sf);
+  int delayRemaining = validRSSIdelayUs - (micros() - rxStartTime);
+  if(delayRemaining > validRSSIdelayUs)
+  {
+    // delayMicroseconds() only valid up to ~16ms per arduino spec. limit to max wait time we need just in case.
+    delayRemaining = validRSSIdelayUs;
+  }
+  if(delayRemaining > 0)
+  {
+    delayMicroseconds(delayRemaining);
+  }
+
```

Comment:
I don't really know how `delayRemaining > validRSSIdelayUs` is possible if delayRemaining is validRSSIdelayUs - elapsed. All of this using ints is a possible overflow, so use all unsigned throughout
```
  uint32_t validRSSIdelayUs = SpreadingFactorToRSSIvalidDelayUs(ExpressLRS_currAirRate_Modparams->sf);
  uint32_t elapsed = micros() - rxStartTime;
  if (elapsed < validRSSIdelayUs)
  {
    delayMicroseconds(validRSSIdelayUs - elapsed);
  }
```

Reply:
Implemented